### PR TITLE
Couple of fixes in "save annotations"

### PR DESCRIPTION
**Pull request by bsekachev**
Related with #949

### CODE REVIEW

File: cvat-core/src/annotations-saver.js [hunk 1]
```diff
@@ -102,12 +102,18 @@
                 },
             };
 
+            const keys = ['id', 'label_id', 'group', 'frame',
```
Comment (nmanovic) [inline cvat-core/src/annotations-saver.js]: Let's create a function to covert an object into "hash". Thus you don't need to repeat keys in several files.

File: cvat/apps/engine/static/engine/js/annotationSaver.js [hunk 2]
```diff
@@ -241,20 +245,33 @@ class AnnotationSaverModel extends Listener {
...
             } else {
                 const [created, updated, deleted] = this._split(exported);
                 this.notify('saveCreated');
                 const savedCreated = await this._create(created);
                 this._updateCreatedObjects(created, savedCreated, mapping);
                 this._version = savedCreated.version;
+
                 for (const type of Object.keys(this._initialObjects)) {
                     for (const object of savedCreated[type]) {
+                        if (object.shapes) {
```
Comment (nmanovic) [inline cvat/apps/engine/static/engine/js/annotationSaver.js]: Does it make sense to create a special function here and use it 3 times?

File: cvat/apps/engine/static/engine/js/annotationSaver.js [hunk 3]
```diff
@@ -241,20 +245,33 @@ class AnnotationSaverModel extends Listener {
                 this._shapeCollection.flush = false;
                 this._version = savedObjects.version;
                 this._resetState();
+
                 for (const type of Object.keys(this._initialObjects)) {
                     for (const object of savedObjects[type]) {
+                        if (object.shapes) {
```
Comment (nmanovic) [inline cvat/apps/engine/static/engine/js/annotationSaver.js]: Could you please add a comment why it is important to delete shape.id here?

File: cvat-core/src/annotations-saver.js (hunk 1 above)
Comment (bsekachev) [reply cvat-core/src/annotations-saver.js]: @nmanovic 
The first file from legacy UI,
The second from cvat-core.

They don't related anyhow.

File: cvat/apps/engine/static/engine/js/annotationSaver.js (hunk 2 above)
Comment (bsekachev) [reply cvat/apps/engine/static/engine/js/annotationSaver.js]: Probably, it does

Comment (nmanovic) [reply cvat/apps/engine/static/engine/js/annotationSaver.js]: I missed that it is from legacy UI. Let's keep it as is for now.

**Pull request merged**